test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Discovery/TypeEnumeratorTests.cs

```diff
@@ -121,6 +122,58 @@ public void GetTestsShouldNotReturnBaseTestMethodsFromAnotherAssembly()
             Assert.IsTrue(methodCount > tests.Count);
         }
 
+        [TestMethod]
+        public void GetTestsShouldReturnBaseTestMethodsFromAnotherAssemblyByConfiguration()
+        {
+            string runSettingxml =
+            @"<RunSettings>   
+                <MSTestV2>
+                  <CaptureTraceOutput>true</CaptureTraceOutput>
+                  <MapInconclusiveToFailed>false</MapInconclusiveToFailed>
+                  <EnableBaseClassTestMethodsFromOtherAssemblies>true</EnableBaseClassTestMethodsFromOtherAssemblies>
+              </MSTestV2>
+            </RunSettings>";
+
+            var mockRunContext = new Mock<IRunContext>();
+            var mockRunSettings = new Mock<IRunSettings>();
+            mockRunContext.Setup(dc => dc.RunSettings).Returns(mockRunSettings.Object);
+            mockRunSettings.Setup(rs => rs.SettingsXml).Returns(runSettingxml);
+
+            MSTestSettings.PopulateSettings(mockRunContext.Object);
+            this.SetupTestClassAndTestMethods(isValidTestClass: true, isValidTestMethod: true);
+            TypeEnumerator typeEnumerator = this.GetTypeEnumeratorInstance(typeof(DummyDerivedFromRemoteTestClass), Assembly.GetExecutingAssembly().FullName);
+
+            var tests = typeEnumerator.Enumerate(out this.warnings);
+
+            Assert.IsTrue(tests.Any(t => t.TestMethod.Name == "BaseTestMethod"));
+        }
+
+        [TestMethod]
+        public void GetTestsShouldNotReturnBaseTestMethodsFromAnotherAssemblyByConfiguration()
+        {
+            string runSettingxml =
+                @"<RunSettings>   
+                <MSTestV2>
+                  <CaptureTraceOutput>true</CaptureTraceOutput>
+                  <MapInconclusiveToFailed>false</MapInconclusiveToFailed>
+                  <EnableBaseClassTestMethodsFromOtherAssemblies>false</EnableBaseClassTestMethodsFromOtherAssemblies>
+              </MSTestV2>
+            </RunSettings>";
+
+            var mockRunContext = new Mock<IRunContext>();
+            var mockRunSettings = new Mock<IRunSettings>();
+            mockRunContext.Setup(dc => dc.RunSettings).Returns(mockRunSettings.Object);
+            mockRunSettings.Setup(rs => rs.SettingsXml).Returns(runSettingxml);
+
+            MSTestSettings.PopulateSettings(mockRunContext.Object);
+            this.SetupTestClassAndTestMethods(isValidTestClass: true, isValidTestMethod: true);
+            TypeEnumerator typeEnumerator = this.GetTypeEnumeratorInstance(typeof(DummyDerivedFromRemoteTestClass), Assembly.GetExecutingAssembly().FullName);
+
+            var tests = typeEnumerator.Enumerate(out this.warnings);
+
+            Assert.IsTrue(tests.All(t => t.TestMethod.Name != "BaseTestMethod"));
```

Comment:
How would that work? tests.Contains would expect an instance of UnitTestElement...